app/src/main/java/com/nutomic/syncthingandroid/views/DevicesAdapter.java

```diff
@@ -45,25 +45,39 @@ public View getView(int position, View convertView, @NonNull ViewGroup parent) {
 
         name.setText(getItem(position).getDisplayName());
         Resources r = getContext().getResources();
-        boolean haveInfo = mConnections != null && mConnections.connections.containsKey(deviceId) &&
-                mConnections.connections.get(deviceId).connected;
-        if (haveInfo) {
-            Connections.Connection conn = mConnections.connections.get(deviceId);
-            if (conn.completion == 100) {
-                status.setText(r.getString(R.string.device_up_to_date));
-                status.setTextColor(ContextCompat.getColor(getContext(), R.color.text_green));
-            }
-            else {
-                status.setText(r.getString(R.string.device_syncing, conn.completion));
-                status.setTextColor(ContextCompat.getColor(getContext(), R.color.text_blue));
+
+        Connections.Connection conn = null;
+        if (mConnections != null) {
+            if (mConnections.connections.containsKey(deviceId)) {
+                conn = mConnections.connections.get(deviceId);
             }
-            download.setText(Util.readableTransferRate(getContext(), conn.inBits));
-            upload.setText(Util.readableTransferRate(getContext(), conn.outBits));
         }
-        else {
+
+        if (conn != null)
+        {
+            download.setText(Util.readableTransferRate(getContext(), (conn.paused || !conn.connected) ? 0 : conn.inBits));
+            upload.setText(Util.readableTransferRate(getContext(), (conn.paused || !conn.connected) ? 0 : conn.outBits));
```

Comment:
Just move these into the relevant branches, the double inline if over two lines is too much.
Also, I assume inBits and outBits might be 0 if paused or disconnected?